docling-parse 2.1.0 fails to compile with qpdf10

We noticed that docling-parse 2.1.0 fails to build from sources with qpdf10 from c9s / RHEL 9. docling-parse<=2.0.5 is building fine.

```
  In file included from /mount/work-dir/docling_parse-2.1.0/docling_parse-2.1.0/src/v2.h:41,
                   from /mount/work-dir/docling_parse-2.1.0/docling_parse-2.1.0/app/parse_v2_fonts.cpp:17:
  /mount/work-dir/docling_parse-2.1.0/docling_parse-2.1.0/src/v2/qpdf/to_json.h: In function 'nlohmann::json_abi_v3_11_3::json pdflib::extract_metadata_in_json(QPDF&, QPDFObjectHandle&)':
  /mount/work-dir/docling_parse-2.1.0/docling_parse-2.1.0/src/v2/qpdf/to_json.h:354:69: error: conversion from 'PointerHolder<Buffer>' to non-scalar type 'std::shared_ptr<Buffer>' requested
    354 |                 std::shared_ptr<Buffer> ptr = metadata.getStreamData(qpdf_dl_all);
        |                                               ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
  gmake[2]: *** [CMakeFiles/parse_v2_fonts.exe.dir/build.make:76: CMakeFiles/parse_v2_fonts.exe.dir/app/parse_v2_fonts.cpp.o] Error 1
  gmake[1]: *** [CMakeFiles/Makefile2:145: CMakeFiles/parse_v2_fonts.exe.dir/all] Error 2
  gmake[1]: *** Waiting for unfinished jobs....
  In file included from /mount/work-dir/docling_parse-2.1.0/docling_parse-2.1.0/src/v2.h:41,
                   from /mount/work-dir/docling_parse-2.1.0/docling_parse-2.1.0/app/parse_v2.cpp:3:
  /mount/work-dir/docling_parse-2.1.0/docling_parse-2.1.0/src/v2/qpdf/to_json.h: In function 'nlohmann::json_abi_v3_11_3::json pdflib::extract_metadata_in_json(QPDF&, QPDFObjectHandle&)':
  /mount/work-dir/docling_parse-2.1.0/docling_parse-2.1.0/src/v2/qpdf/to_json.h:354:69: error: conversion from 'PointerHolder<Buffer>' to non-scalar type 'std::shared_ptr<Buffer>' requested
    354 |                 std::shared_ptr<Buffer> ptr = metadata.getStreamData(qpdf_dl_all);
        |                                               ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
  gmake[2]: *** [CMakeFiles/parse_v2.exe.dir/build.make:76: CMakeFiles/parse_v2.exe.dir/app/parse_v2.cpp.o] Error 1
  gmake[1]: *** [CMakeFiles/Makefile2:119: CMakeFiles/parse_v2.exe.dir/all] Error 2
```

--- a/src/v2/qpdf/to_json.h
+++ b/src/v2/qpdf/to_json.h
@@ -351,7 +351,7 @@ namespace pdflib
 
 	    if(metadata.isStream())
 	      {
-		std::shared_ptr<Buffer> ptr = metadata.getStreamData(qpdf_dl_all);
+		auto ptr = metadata.getStreamData(qpdf_dl_all);
 		
 		// Convert raw data to std::string
 		std::string content(reinterpret_cast<const char*>(ptr->getBuffer()), ptr->getSize());

